core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -55,4 +56,28 @@ private SnapshotUtil() {
     }
     return ancestorIds;
   }
+
+  public static boolean isCurrentAncestor(TableMetadata tableMeta, long snapshotId) {
+    List<Long> currentAncestors = SnapshotUtil.ancestorIds(tableMeta.currentSnapshot(), tableMeta::snapshot);
+    return currentAncestors.contains(snapshotId);
+  }
+
+  /**
+   * Return the latest snapshot whose timestamp is before the provided timestamp.
+   * @param tableMeta TableMetadata representing the table state on which the snapshot is being looked up
+   * @param timestampMillis lookup snapshots before this timestamp
+   * @return
+   */
+  public static Snapshot findLatestSnapshotOlderThan(TableMetadata tableMeta, long timestampMillis) {
+    long snapshotTimestamp = 0;
+    Snapshot result = null;
+    for (Snapshot snapshot : tableMeta.snapshots()) {
```

Comment:
Should this traverse `ancestorIds` instead? The reason is that `rollbackTo` checks that the target snapshot is an ancestor of the current table state to verify it is a rollback and not a fast-forward or a complete change. Since this is called only by `rollbackToTime`, I think it makes sense to have the same requirement.

If you agree, I'd have this traverse the ancestor IDs and rename it to `findLatestAncestorOlderThan`.

Reply:
makes sense.. otherwise  `rollbackToTime()` behaviour seems inconsistent from `rollbackTo()`